### PR TITLE
MES-3631 - Enforce Minimum App Version

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Now pass app_version as a query string parameter, when requesting app config
- Display a message when the app_version is below the minimum version. 

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)

IMG_CE192FAB610A-1

### CODE REVIEW

File: src/providers/app-config/app-config.ts [hunk 1]
```diff
@@ -135,33 +140,43 @@ export class AppConfigProvider {
...
+          const url = `${this.environmentFile.configUrl}?app_version=${version}`;
+          this.httpClient.get<any>(url)
+            .pipe(timeout(30000))
+            .subscribe(
+              (data) => {
+                this.dataStore.setItem('CONFIG', JSON.stringify(data));
+                resolve(data);
+              },
+              (error: HttpErrorResponse) => {
+                if (this.shouldGetCachedConfig(error.error)) {
```
Comment (buzzub) [inline src/providers/app-config/app-config.ts:154]: `error.error` 😆

**Review by buzzub — APPROVED** [review]

File: src/providers/app-config/app-config.ts [hunk 2]
```diff
@@ -135,33 +140,43 @@ export class AppConfigProvider {
...
       } else {
         this.getCachedRemoteConfig()
           .then(data => resolve(data))
           .catch(error => reject(error));
       }
     })
 
+  private shouldGetCachedConfig = (errorMessage: string): boolean =>
+    errorMessage  !== AuthenticationError.USER_NOT_AUTHORISED &&
+    errorMessage !== AppConfigError.INVALID_APP_VERSION
```
Comment (rossfauldsbjss) [inline src/providers/app-config/app-config.ts:175]: I know it's valid syntax - but i thought coding standards are to have ; on the end... (and i thought there were linting rules around this)

File: src/pages/login/__tests__/login.spec.ts [hunk 3]
```diff
@@ -377,5 +385,15 @@ describe('LoginPage', () => {
...
+    it('should show the correct div if user has an old version of the app', () => {
+      component.hasUserLoggedOut = false;
+      component.appInitError = AppConfigError.INVALID_APP_VERSION;
+      fixture.detectChanges();
+
+      const tags = fixture.debugElement.queryAll(By.css('h2'));
+      expect(tags.length).toBe(1);
+      expect((tags[0].nativeElement as HTMLElement).textContent).toContain('Unsupported application version');
+    });
+
```
Comment (rossfauldsbjss) [inline src/pages/login/__tests__/login.spec.ts:397]: Would like to see an explicit test that the div is not there if app version is okay.

**Review by rossfauldsbjss — APPROVED** [review]
looks good. couple of minor suggestions,

File: src/providers/app-config/app-config.ts (hunk 2 above)
Comment (TomBillingtonUK) [reply src/providers/app-config/app-config.ts:175]: liniting told me to take of the ;

File: src/pages/login/__tests__/login.spec.ts (hunk 3 above)
Comment (TomBillingtonUK) [reply src/pages/login/__tests__/login.spec.ts:397]: will add now :)

**Pull request merged**